## Setup

In [ ]:
import pandas as pd

In [ ]:
ufos_path = 'YOUR PATH HERE'
sightings = pd.read_csv(ufos_path)
sightings.head()

<ipython-input-3-828416dd11ab>:2: DtypeWarning: Columns (5,9) have mixed types. Specify dtype option on import or set low_memory=False.
  sightings = pd.read_csv(ufos_path)


datetime                  city state country     shape  \
0  10/10/1949 20:30            san marcos    tx      us  cylinder   
1  10/10/1949 21:00          lackland afb    tx     NaN     light   
2  10/10/1955 17:00  chester (uk/england)   NaN      gb    circle   
3  10/10/1956 21:00                  edna    tx      us    circle   
4  10/10/1960 20:00               kaneohe    hi      us     light   

  duration (seconds) duration (hours/min)  \
0               2700           45 minutes   
1               7200              1-2 hrs   
2                 20           20 seconds   
3                 20             1/2 hour   
4                900           15 minutes   

                                            comments date posted    latitude  \
0  This event took place in early fall around 194...   4/27/2004  29.8830556   
1  1949 Lackland AFB&#44 TX.  Lights racing acros...  12/16/2005    29.38421   
2  Green/Orange circular disc over Chester&#44 En...   1/21/2008        53.2   
3  My older brother and twin sister were leaving ...   1/17/2004  28.9783333   
4  AS a Marine 1st Lt. flying an FJ4B fighter/att...   1/22/2004  21.4180556   

   longitude  
0      -97.9  
1      -98.6  
2       -2.9  
3      -96.6  
4     -157.8

## Use "and" logic to generate a filtered subset of UFO sightings that meet the following criteria:

* The sightings must have occurred in one of the following states: Kentucky (abbreviated "ky"), Tennessee ("tn"), or West Virginia ("wv").

* AND, the sightings must have been 60 seconds or less in duration.

In [ ]:
sightings['duration (seconds)'] = pd.to_numeric(sightings['duration (seconds)'], errors='coerce')

In [ ]:
ky_tn_wv = sightings['state'].isin(['ky','tn','wv'])
lte_60_sec = sightings['duration (seconds)'] <= 60

subset_1 = sightings[ky_tn_wv & lte_60_sec]
subset_1.head(10)

datetime          city state country     shape  \
26   10/10/1977 22:00    louisville    ky      us     light   
53   10/10/1992 20:15       seymour    tn      us     cigar   
57   10/10/1993 23:00      carthage    tn      us     other   
81   10/10/1998 23:50        delano    tn      us  fireball   
99   10/10/2000 21:30      florence    ky      us     light   
133  10/10/2004 14:00    morgantown    wv      us    circle   
281  10/10/2013 21:17    lost creek    wv      us  triangle   
299  10/11/1998 14:30  mount carmel    tn      us     light   
340  10/11/2003 07:00     fort knox    ky     NaN  fireball   
375  10/11/2006 01:25     nashville    tn      us      disk   

     duration (seconds) duration (hours/min)  \
26                 30.0   approx: 30 seconds   
53                 60.0            1min. 39s   
57                 60.0      less than 1 min   
81                 15.0           15 seconds   
99                  5.0          4-5 seconds   
133                10.0               10 sec   
281                45.0           45 seconds   
299                10.0           10 seconds   
340                 3.0          2-3 seconds   
375                13.0               13 sec   

                                              comments date posted  \
26   HBCCUFO CANADIAN REPORT:  Pilot Sighting Of Un...   3/17/2004   
53   Stationary Elongated UFO 200ft above vacant fi...  10/31/2008   
57                  1 object with green and red lights   3/21/2003   
81   I came home from work one night and I seen a b...   3/21/2003   
99   Two objects traveling side by side pass over&#...   12/2/2000   
133  Solid round silver ball passing over at about ...   5/24/2005   
281  Small triangle with 3 lights floating low over...  10/14/2013   
299  Bright shiny object splits in two and disappea...   11/1/1998   
340  i saw a large fireball falling to earth in the...   11/8/2003   
375  Circle shaped solid white object flying throug...  10/30/2006   

       latitude  longitude  
26   38.2541667      -85.8  
53   35.8905556      -83.7  
57   36.2522222      -86.0  
81   35.2650000      -84.6  
99   38.9988889      -84.6  
133  39.6294444      -80.0  
281  39.1611111      -80.4  
299  36.5452778      -82.7  
340   37.916104      -86.0  
375  36.1658333      -86.8

## How many sightings met these criteria?

In [ ]:
len(subset_1)

908

**908**

## Use "and" logic to generate a filtered subset of UFO sightings that meet the following criteria:

* The sightings must have been over 10 minutes in duration.

* AND, the sightings must have occurred in the United States (abbreviated "us").

* AND, the sightings must have occurred between 33 and 38 degrees latitude.

In [ ]:
sightings['latitude'] = pd.to_numeric(sightings['latitude'], errors='coerce')

In [ ]:
gt_10_min = sightings['duration (seconds)'] > 600
in_us = sightings['country'] == 'us'
bt_33_38_lat = sightings['latitude'].between(33,38)

subset_2 = sightings[gt_10_min & in_us & bt_33_38_lat]
subset_2.head(10)

datetime            city state country      shape  \
15   10/10/1972 19:00   harlan county    ky      us     circle   
49   10/10/1991 22:00          frisco    nc      us    unknown   
76   10/10/1998 17:30       las vegas    nv      us     circle   
88   10/10/1999 21:00          rachel    nv      us      light   
107  10/10/2001 21:30          fresno    ca      us   changing   
123  10/10/2003 21:10  crescent beach    sc      us  formation   
166  10/10/2007 01:00     stockbridge    ga      us   changing   
175  10/10/2007 20:30         conyers    ga      us    unknown   
206  10/10/2009 19:30         suffolk    va      us    unknown   
226  10/10/2010 21:00      kykotsmovi    az      us       disk   

     duration (seconds) duration (hours/min)  \
15               1200.0            20minutes   
49               1800.0           30 minutes   
76               2700.0               45min.   
88              10800.0              3 hours   
107               900.0        15 min. apprx   
123             37800.0            1 1/2 hr.   
166              3600.0               1 hour   
175              3600.0               1 hour   
206               900.0           15 minutes   
226              7200.0              2 hours   

                                              comments date posted  latitude  \
15   On october 10&#44 1972 myself&#44my 5yrs.daugh...   9/15/2005      36.8   
49   A friend and myself were standing on the shore...   1/28/1999      35.2   
76               Ufo sighting in las vegas near Area51   8/30/1999      36.2   
88   Bright lights with incredible agility seen fro...   5/24/2005      37.6   
107  Objects were sighted driving north on Highway ...  11/20/2001      36.7   
123  For two consecutive nights&#44 we watched a pa...   1/17/2004      33.8   
166                                     i was abducted  11/28/2007      33.5   
175  Craft seen falling and dissapears into thin ai...  11/28/2007      33.7   
206  It looked sort of lime green.  It appeared to ...  12/12/2009      36.7   
226  We saw  what seem to be a space craft in the e...   1/31/2011      35.9   

     longitude  
15       -83.3  
49       -75.6  
76      -115.1  
88      -115.7  
107     -119.8  
123      -78.7  
166      -84.2  
175      -84.0  
206      -76.6  
226     -110.6

## How many sightings met these criteria?

In [ ]:
len(subset_2)

4449

**4449**